app/src/main/java/com/duckduckgo/app/browser/applinks/AppLinksLauncher.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.applinks
+
+import android.content.ActivityNotFoundException
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.widget.Toast
+import androidx.annotation.RequiresApi
+import androidx.annotation.StringRes
+import com.duckduckgo.app.browser.BrowserTabViewModel
+import com.duckduckgo.app.browser.R
+import com.duckduckgo.app.browser.SpecialUrlDetector.UrlType.AppLink
+import com.duckduckgo.appbuildconfig.api.AppBuildConfig
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+import timber.log.Timber
+
+interface AppLinksLauncher {
+    fun openAppLink(context: Context?, appLink: AppLink, viewModel: BrowserTabViewModel)
```

Comment:
I was wondering if we can remove the `viewModel: BrowserTabViewModel` from here and use `AppLinksHandler` in `DuckDuckGoAppLinksLauncher`.

Something like:
`fun openAppLink(context: Context?, appLink: AppLink)` 

and in `DuckDuckGoAppLinksLauncher` to use `appLinksHandler.updatePreviousUrl(null)` instead of `viewModel.clearPreviousUrl()`.

Reply:
This isn't possible unfortunately because it needs the same instance of `appLinksHandler` used in the `ViewModel`